crates/runtime/src/component/mod.rs

```diff
@@ -37,3 +154,462 @@ fn claims(wasm: impl AsRef<[u8]>) -> Result<Option<jwt::Claims<jwt::Component>>>
     ensure!(v.signature_valid, "signature is not valid");
     Ok(Some(claims.claims))
 }
+
+/// Pre-compiled component [Component], which is cheapily-[Cloneable](Clone)
+#[derive(Clone)]
+pub struct Component<H>
+where
+    H: Handler,
+{
+    engine: wasmtime::Engine,
+    claims: Option<jwt::Claims<jwt::Component>>,
+    instance_pre: wasmtime::component::InstancePre<Ctx<H>>,
+    max_execution_time: Duration,
+}
+
+impl<H> Debug for Component<H>
+where
+    H: Handler,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Component")
+            .field("claims", &self.claims)
+            .field("runtime", &"wasmtime")
+            .field("max_execution_time", &self.max_execution_time)
+            .finish_non_exhaustive()
+    }
+}
+
+fn type_annotate_http<F, H>(val: F) -> F
+where
+    H: Handler,
+    F: Fn(&mut Ctx<H>) -> WasiHttpImpl<&mut Ctx<H>>,
+{
+    val
+}
+
+fn new_store<H: Handler>(
+    engine: &wasmtime::Engine,
+    handler: H,
+    max_execution_time: Duration,
+) -> wasmtime::Store<Ctx<H>> {
+    let table = ResourceTable::new();
+    let wasi = WasiCtxBuilder::new()
+        .args(&["main.wasm"]) // TODO: Configure argv[0]
```

Comment:
Any work needed here or is it ok to just have an issue?

Reply:
This was just moved from existing implementation, IMO this is fine, but I can open an issue

Reply:
https://github.com/wasmCloud/wasmCloud/issues/2476